### PR TITLE
Add AprilTagVision example

### PR DESCRIPTION
**Pull request by Suave101**
A task within issue #49. This is a port of the Java AprilTagsVision example to Python

### CODE REVIEW

**Comment by Suave101** [comment]
This is a Draft pull request for a couple of reasons:
1. This is my first pull request and I am not sure if all the formatting is perfect
2. I have tested this code but, I am not confident in my test due to the following quote from the robotpy documentation:

> "The child process will NOT be launched when running the robot code in simulation or unit testing mode", [RobotPy Documentation](https://robotpy.readthedocs.io/en/stable/vision/roborio.html#important-notes)

File: AprilTagsVision/Robot.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env python3
```
Comment (virtuald) [inline AprilTagsVision/Robot.py:1]: robot.py should be lowercase `robot.py`

File: AprilTagsVision/Robot.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+# the WPILib BSD license file in the root directory of this project.
+#
+
+
+import wpilib
+import wpilib.cameraserver
+
+
+#
+# This is a demo program showing the detection of AprilTags. The image is acquired from the USB
```
Comment (virtuald) [inline AprilTagsVision/Robot.py:14]: This needs to be a docstring for `MyRobot`

File: AprilTagsVision/vision.py [hunk 3]
```diff
@@ -0,0 +1,156 @@
+#!/usr/bin/env python3
```
Comment (virtuald) [inline AprilTagsVision/vision.py:1]: I don't have time to test this at the moment, but if you want to try it on your computer then just install `robotpy[cscore]` and `opencv-python` and you can run this file locally with your computer's camera via `python -m cscore vision.py`.

Probably should add a comment about that here too.

**Review by virtuald — CHANGES_REQUESTED** [review]

**Comment by Suave101** [comment]
A test of the vision.py file has been conducted and bugs were found. The bugs were subsequently fixed and code reformatted into black. The `robot.py` file has been changed to lowercase and the comment at the beginning of the `robot.py` has been changed to a docstring!

**Comment by Suave101** [comment]
The following workflow does not allow the first line of vision.py to be `#!/usr/bin/env python3`:
https://github.com/robotpy/examples/actions/runs/7383683670/job/20099069304

**Comment by Suave101** [comment]
Could someone please help me fix what is causing the failure in checks?

**Comment by auscompgeek** [comment]
The new robot is missing from `run_tests.sh`.

File: AprilTagsVision/Robot.py (hunk 1 above)
Comment (Suave101) [reply AprilTagsVision/Robot.py]: Done
https://github.com/robotpy/examples/pull/100#issuecomment-1873734226

File: AprilTagsVision/Robot.py (hunk 2 above)
Comment (Suave101) [reply AprilTagsVision/Robot.py]: Done:
https://github.com/robotpy/examples/pull/100#issuecomment-1873734226

File: AprilTagsVision/vision.py (hunk 3 above)
Comment (Suave101) [reply AprilTagsVision/vision.py]: Done!

File: AprilTagsVision/robot.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+# Open Source Software; you can modify and/or share it under the terms of
+# the WPILib BSD license file in the root directory of this project.
+#
+
+
+import wpilib
+import wpilib.cameraserver
+
+
+"""
```
Comment (auscompgeek) [inline AprilTagsVision/robot.py:13]: This isn't a docstring – it's not the first statement of a module/class/function.

**Comment by virtuald** [comment]
Please rebase and remove wpilib.run (see https://github.com/robotpy/examples/pull/106)

**Comment by Suave101** [comment]
Do I close this pull and rebase this branch into a new branch with a rebase pull request then open a new pull. As well as removing wpilib.run as per #106?

**Comment by Suave101** [comment]
Closing for Rebase

**Comment by virtuald** [comment]
When you rebase, you can force push your branch that way you don't have to recreate the PR.

**Pull request closed without merging**